### PR TITLE
CLI add/remove registry commands

### PR DESCRIPTION
**Pull request by aloubyansky**
This PR adds

* `quarkus registry add <registry-id>[,<registry-id>] [--config <path-to-config>]`
* `quarkus registry remove <registry-id>[,<registry-id>] [--config <path-to-config>]`

commands to the CLI.

### CODE REVIEW

**Comment by aloubyansky** [comment]
fyi @quarkusio/devtools

File: devtools/cli/src/main/java/io/quarkus/cli/RegistryAddCommand.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import picocli.CommandLine;
+
+@CommandLine.Command(name = "add", sortOptions = false, showDefaultValues = true, mixinStandardHelpOptions = false, header = "Add a Quarkus extension registry to the registry client configuration", description = "%n"
+        + "This command will add a Quarkus extension registry to the registry client configuration unless it's already present", headerHeading = "%n", commandListHeading = "%nCommands:%n", synopsisHeading = "%nUsage: ", parameterListHeading = "%n", optionListHeading = "Options:%n")
+public class RegistryAddCommand extends BaseRegistryCommand {
+
+    @CommandLine.Mixin
+    protected RegistryClientMixin registryClient;
+
+    @CommandLine.Parameters(arity = "0..1", paramLabel = "REGISTRY-ID[,REGISTRY-ID]", description = "Registry ID to add to the registry client configuration")
```
Comment (ebullient) [inline devtools/cli/src/main/java/io/quarkus/cli/RegistryAddCommand.java:26]: Add an example value to the description, e.g. https://github.com/quarkusio/quarkus/blob/fb575c117f6a4958bcf2a1f21a134fc7b98b1198/devtools/cli/src/main/java/io/quarkus/cli/create/TargetGAVGroup.java#L15

File: devtools/cli/src/main/java/io/quarkus/cli/RegistryAddCommand.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    String registryIds;
+
+    @CommandLine.Option(paramLabel = "CONFIG", names = { "--config" }, description = "Configuration file")
```
Comment (ebullient) [inline devtools/cli/src/main/java/io/quarkus/cli/RegistryAddCommand.java:29]: Does this configuration parameter belong on the RegistryClientMixin instead? (such that any command involving the registry could be instructed to use that config?)

File: devtools/cli/src/main/java/io/quarkus/cli/RegistryRemoveCommand.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import picocli.CommandLine;
+
+@CommandLine.Command(name = "remove", sortOptions = false, showDefaultValues = true, mixinStandardHelpOptions = false, header = "Remove a Quarkus extension registry from the registry client configuration", description = "%n"
+        + "This command will remove a Quarkus extension registry from the registry client configuration", headerHeading = "%n", commandListHeading = "%nCommands:%n", synopsisHeading = "%nUsage: ", parameterListHeading = "%n", optionListHeading = "Options:%n")
+public class RegistryRemoveCommand extends BaseRegistryCommand {
+
+    @CommandLine.Mixin
+    protected RegistryClientMixin registryClient;
+
+    @CommandLine.Parameters(arity = "0..1", paramLabel = "REGISTRY-ID[,REGISTRY-ID]", description = "Registry ID to remove from the registry client configuration")
```
Comment (ebullient) [inline devtools/cli/src/main/java/io/quarkus/cli/RegistryRemoveCommand.java:25]: Same as for add, show an example registry id in the description.

File: devtools/cli/src/main/java/io/quarkus/cli/RegistryRemoveCommand.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    String registryIds;
+
+    @CommandLine.Option(paramLabel = "CONFIG", names = { "--config" }, description = "Configuration file")
```
Comment (ebullient) [inline devtools/cli/src/main/java/io/quarkus/cli/RegistryRemoveCommand.java:28]: Same as for add: does this parameter belong in the RegistryClientMixin

**Review by ebullient — CHANGES_REQUESTED** [review]

File: devtools/cli/src/main/java/io/quarkus/cli/build/GradleRunner.java [hunk 5]
```diff
@@ -225,7 +225,9 @@ void setGradleProperties(ArrayDeque<String> args, boolean batchMode) {
         }
         args.add(registryClient.getRegistryClientProperty());
 
-        final String configFile = System.getProperty(RegistriesConfigLocator.CONFIG_FILE_PATH_PROPERTY);
+        final String configFile = registryClient.getConfigArg() == null
```
Comment (ebullient) [inline devtools/cli/src/main/java/io/quarkus/cli/build/GradleRunner.java:228]: Should RegistryClientMixin be checking/handling this config file property for all cases (not just Gradle?)

**Review by ebullient — CHANGES_REQUESTED** [review]

File: devtools/cli/src/main/java/io/quarkus/cli/build/GradleRunner.java (hunk 5 above)
Comment (aloubyansky) [reply devtools/cli/src/main/java/io/quarkus/cli/build/GradleRunner.java:228]: It's not necessary for Maven. It could be for jbang.

**Review by ebullient — APPROVED** [review]

File: devtools/cli/src/main/java/io/quarkus/cli/build/GradleRunner.java (hunk 5 above)
Comment (ebullient) [reply devtools/cli/src/main/java/io/quarkus/cli/build/GradleRunner.java:228]: We should look at sorting this out after... either this property is around or it isn't...

**Pull request merged**